x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elastic/authorization/ElasticInferenceServiceAuthorizationModel.java

```diff
@@ -205,6 +206,19 @@ private static void validateConfigurationForTextEmbedding(ElasticInferenceServic
             config.similarity(),
             TaskType.TEXT_EMBEDDING
         );
+
+        var configElementType = config.elementType().toLowerCase(Locale.ROOT);
+        var supportedElementType = getSupportedElementType();
+
+        if (configElementType.equals(supportedElementType) == false) {
+            throw new IllegalArgumentException(
+                Strings.format("Unsupported element type encountered [%s], only [%s] is supported", configElementType, supportedElementType)
+            );
+        }
+    }
+
+    private static String getSupportedElementType() {
+        return ElasticInferenceServiceDenseTextEmbeddingsServiceSettings.SUPPORTED_ELEMENT_TYPE.toString().toLowerCase(Locale.ROOT);
```

Comment:
Would it be better to have this be `getSupportedElementTypes()` and return a Set? There's only one supported element type now, but if more are added in the future it'll make it easier to accommodate that without having to change the code or exception message.